### PR TITLE
Default file name in fileinfo

### PR DESCRIPTION
**Pull request by eprikazc**
I have created new pull request based on feedback from https://github.com/amol-/depot/pull/21
Here I moved FileStorage.fileinfo to separate class as it seemed that it was accumulating too much logic to fit to single function.

### CODE REVIEW

**Comment by amol-** [comment]
``FileStorage.fileinfo`` was a public API that people might have used in their code, we cannot throw it away without a `DeprecationWarning` and a deprecation phase. It should at least wrap the `FileInfo` class and provide backward compatibility with tests that ensure backward compatibility for a few releases.

Also a class that you create just to call a single method should probably just be a function :)
See https://www.youtube.com/watch?v=o9pEzgHorH0 for a nice discussion around that topic.

By the way, my only request is that `FileInfo` becames an internal class (`_FileInfo`) used only by ``FileStorage.fileinfo`` and everything else keeps working around the ``FileStorage.fileinfo`` method. That will give us time to deprecate it in the future while and think a clean and long-term interface for `FileInfo` class while being able to merge the change without too much changes.

In the future I suggest you try to keep a single change in each pull request, having a single huge pull request that: adds a new feature, changes an interface and removes a public api changing 5-6 classes using it makes pretty hard to check the pull request and merge it safely. It would have been easier to manage doing one step at time in different pull requests :D

File: depot/io/local.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def __init__(self, file_id, local_path):
         except:
             raise IOError('File %s not existing' % file_id)
 
-        metadata_info = {'filename': 'unknown',
+        metadata_info = {'filename': 'unnamed',
```
Comment (eprikazc) [inline depot/io/local.py]: Shouldn't we asume that filename and content_type are always in metadata? Why repeat here default values? 
Is it for backwards compatibility?

**Comment by eprikazc** [comment]
Hey Alessandro, thanks for the feedback :)

I have moved FileInfo to FileStorage._FileInfo and kept FileStorage.fileinfo to not break backwards compatibility. Now fileinfo just has extra parameter to be used in "replace" cases. I am open to refactoring _FileInfo.

I agree with you - it is better not to put so many changes in single pull request. If you wish I can rollback "unknown" => "unnamed" renaming and do it in separate pull request.

**Comment by amol-** [comment]
Reopening as I want to merge this, just didn't have time to apply some minor tweaks I would like to have before merging it.

**Pull request merged**